text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,384 @@
+  impossible, as they would likely lead to widespread replacement of existing resources (typically, because a new
+  logical ID would be assigned to those):
+
+  | Decision | RFC / Issue    | Description                                                           |
+  | -------- | -------------- | --------------------------------------------------------------------- |
+  | 🚫 Out   | [aws-cdk#1687] | Un-mangle logical IDs generated for high-level (a.k.a. L2) constructs |
+  | 🚫 Out   | [aws-cdk#6421] | Change logical ID attribution to avoid potential collisions           |
+
+[rfc-55]: ./0055-feature-flags.md
+[rfc-6]: ./0006-monolothic-packaging.md
+[#39]: https://github.com/aws/aws-cdk-rfcs/issues/39
+[#116]: https://github.com/aws/aws-cdk-rfcs/issues/116
+[aws-cdk#1687]: https://github.com/aws/aws-cdk/issues/1687
+[aws-cdk#3203]: https://github.com/aws/aws-cdk/issues/3203
+[aws-cdk#3930]: https://github.com/aws/aws-cdk/issues/3930
+[aws-cdk#6421]: https://github.com/aws/aws-cdk/issues/6421
+[aws-cdk#6966]: https://github.com/aws/aws-cdk/issues/6966
+
+# Detailed Design
+
+## Overview
+
+The development and release of v2 will be done in several key steps, detailed in the following sections:
+
+1. [Create a v2 development branch by forking v1](#forking-v2-out-of-v1)
+1. [Make the necessary code changes to implement all features scoped in for v2](#feature-implementation)
+1. [Prepare documentation updates](#documentation-updates)
+1. [Publish and advertise _experimental_ pre-releases to the usual package registries](#prerelease-publishing)
+1. [Perform miscelaneous construct library maintenance tasks](#construct-library-maintenance-tasks)
+1. [Announce developer preview and document v1 maintenance plan](#developer-preview)
+1. [Listen to customer feedback and adjust as needed](#feedback-phase)
+1. [Announce General Availability](#general-availability)
+1. [Clean up dead code](#clean-up)
+1. [Monitor adoption of the new version](#monitoring-adoption)
+1. [Deprecation of v1](#deprecation-of-v1)
+
+## Forking v2 out of v1
+
+The forking point will be achieved by cutting a new `next` branch from the current tip of the `master` branch. The
+repository-wide `version` configuration will be updated to `2.0.0-alpha` (the pre-release identifier could be
+different). A continuous integration pipeline will be configured on this new branch, so that pull requests can be
+automatically validated.
+
+As long as the codebases have not diverged too much, it should be possible to continue forward-porting new developments
+on top of the `next` branch by simply cherry-picking new commits from the `master` branch. Alternatively, it might be
+possible to simply periodically merge `master` into `next`.
+
+As the codebases grow further appart, manual work may be required in order to forward-port feature work merged in
+`master`. As a consequence we will strive to keep divergences to a minimum for as long as possible.
```

Comment:
This is going to be quite a hassle, so we need to set up some ownership around it. It won't happen on volunteer basis.